src/core/src/op/slice_scatter.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/op/slice_scatter.hpp"
+
+#include "bound_evaluate.hpp"
+#include "itt.hpp"
+#include "openvino/core/validation_util.hpp"
+#include "slice_scatter_shape_inference.hpp"
+
+namespace ov {
+namespace op {
+namespace v15 {
+SliceScatter::SliceScatter(const Output<Node>& data,
+                           const Output<Node>& updates,
+                           const Output<Node>& start,
+                           const Output<Node>& stop,
+                           const Output<Node>& step)
+    : Op({data, updates, start, stop, step}) {
+    constructor_validate_and_infer_types();
+}
+
+SliceScatter::SliceScatter(const Output<Node>& data,
+                           const Output<Node>& updates,
+                           const Output<Node>& start,
+                           const Output<Node>& stop,
+                           const Output<Node>& step,
+                           const Output<Node>& axes)
+    : Op({data, updates, start, stop, step, axes}) {
+    constructor_validate_and_infer_types();
+}
+
+void SliceScatter::validate_and_infer_types() {
+    OV_OP_SCOPE(v15_SliceScatter_validate_and_infer_types);
+    for (size_t i = 2; i < get_input_size(); ++i) {
+        NODE_VALIDATION_CHECK(this,
+                              get_input_element_type(i).is_integral_number(),
```

Comment:
Dynamic element type should be allowed
```suggestion
                              get_input_element_type(i).is_dynamic() || get_input_element_type(i).is_integral_number(),
```

Reply:
Done + added tests with dynamic types